samples/bluetooth/iso_broadcast_benchmark/src/broadcaster.c

```diff
@@ -221,6 +241,152 @@ static int parse_sdu_arg(void)
 	return (int)sdu;
 }
 
+#if defined(CONFIG_BT_ISO_ADVANCED)
+static int parse_irc_arg(void)
+{
+	size_t char_count;
+	char buffer[4];
+	uint64_t irc;
+
+	printk("Set IRC (current %u, default %u)\n",
+	       big_create_param.irc, DEFAULT_BIS_IRC);
+
+	char_count = get_chars(buffer, sizeof(buffer) - 1);
+	if (char_count == 0) {
+		return DEFAULT_BIS_IRC;
+	}
+
+	irc = strtoul(buffer, NULL, 0);
+	if (irc > BT_ISO_IRC_MAX) {
+		printk("Invalid IRC %llu", irc);
+
+		return -EINVAL;
+	}
+
+	return (int)irc;
+}
+
+static int parse_pto_arg(void)
+{
+	size_t char_count;
+	char buffer[4];
+	uint64_t pto;
+
+	printk("Set PTO (current %u, default %u)\n",
+	       big_create_param.pto, DEFAULT_BIS_PTO);
+
+	char_count = get_chars(buffer, sizeof(buffer) - 1);
+	if (char_count == 0) {
+		return DEFAULT_BIS_PTO;
+	}
+
+	pto = strtoul(buffer, NULL, 0);
+	if (pto > BT_ISO_PTO_MAX) {
+		printk("Invalid PTO %llu", pto);
+
+		return -EINVAL;
+	}
+
+	return (int)pto;
+}
+
+static int parse_iso_interval_arg(void)
+{
+	uint64_t iso_interval;
+	size_t char_count;
+	char buffer[8];
+
+	printk("Set ISO interval (current %u, default %u)\n",
+	       big_create_param.iso_interval, DEFAULT_BIS_ISO_INTERVAL);
+
+	char_count = get_chars(buffer, sizeof(buffer) - 1);
+	if (char_count == 0) {
+		return DEFAULT_BIS_ISO_INTERVAL;
+	}
+
+	iso_interval = strtoul(buffer, NULL, 0);
+	if (iso_interval > BT_ISO_ISO_INTERVAL_MAX) {
+		printk("Invalid ISO interval %llu", iso_interval);
+
+		return -EINVAL;
+	}
+
+	return (int)iso_interval;
+}
+
+static int parse_nse_arg(void)
+{
+	uint64_t num_subevents;
+	size_t char_count;
+	char buffer[4];
+
+	printk("Set number of subevents (current %u, default %u)\n",
+	       bis_iso_qos.num_subevents, DEFAULT_BIS_NSE);
+
+	char_count = get_chars(buffer, sizeof(buffer) - 1);
+	if (char_count == 0) {
+		return DEFAULT_BIS_NSE;
+	}
+
+	num_subevents = strtoul(buffer, NULL, 0);
+	if (num_subevents > BT_ISO_NSE_MAX) {
```

Comment:
there is no check to verify that the NSE is at least BT_ISO_NSE_MIN. In other words it is possible to enter 0 for the number of subevents. Same for the other parameters

Reply:
Added checks for minimum values for both this and the connected ISO benchmark sample